pkg/operator/ceph/object/objectstore.go

```diff
@@ -394,55 +394,48 @@ func getZoneEndpoints(objContext *Context, serviceEndpoint string) ([]string, bo
 	return zoneEndpointsList, isEndpointAlreadyExists, nil
 }
 
+func createMultisiteConfigurations(objContext *Context, configType, configTypeArg string, args ...string) error {
+	args = append([]string{configType}, args...)
+	args = append(args, configTypeArg)
+	// get the multisite config before creating
+	output, getconfigErr := RunAdminCommandNoMultisite(objContext, true, configType, "get", configTypeArg)
+	if getconfigErr != nil {
+		// ENOENT means "No such file or directory"
+		if code, err := exec.ExtractExitCode(getconfigErr); err == nil && code == int(syscall.ENOENT) {
+			// create the object if it doesn't exist yet
+			output, err := RunAdminCommandNoMultisite(objContext, false, args...)
+			if err != nil {
+				return errorOrIsNotFound(err, "failed to create ceph %q %q, for reason %q", configType, configTypeArg, output)
+			}
+			logger.Debugf("created %q %q", configType, configTypeArg)
+		} else {
```

Comment:
This triggers under both of these conditions:
- `exec.ExtractExitCode()`'s err != nil
- code != ENOENT

The first condition is something where our error message being returned could provide too little info for debugging if ExtractExitCode() fails. We should track that specifically. Let's rewrite more of this function to improve its readability (now that it is a function) and track that error specifically so that we are sure we're not leaving corner cases where it's harder to debug.


As an initial note, putting a lot of code in an `if` block is a code smell. If at all possible, keep `if` blocks short, with no nested conditions/loops. A good technique to do that is to try to use the inverse conditional once we notice a long conditional branch. We can do that here by returning success if `getConfigErr == nil`.

We should also examine the use of `errorOrIsNotFound()` more closely to make sure any changes here aren't un-doing the intent of that helper. 

Secondly, let's give a more specific error if `ExtractExitCode()` fails.

```go
output, getConfigErr := 
if getConfigErr == nil {
    return nil # already exists
}
if kerrors.IsNotFound(getConfigErr) {
    // the pod used to exec command (act as a proxy) is not found/ready yet
    // caller can nicely handle error and not overflow logs with misleading error messages
    return getConfigErr
}

code, err := exec.ExtractExitCode(getConfigErr)
if err != nil {
    return errors.Wrapf(getConfigErr, "'radosgw-admin %q get' failed with unknown exit code (%v), for reason %q", configType, err, output)
}

// ENOENT means "No such file or directory" -> resource doesn't exist
if code != int(syscall.ENOENT) {
    # errorOrIsNotFound() unnecessary b/c we already know it's not an IsNotFound err
    errors.Wrapf(getconfigErr, "'radosgw-admin %q get' failed with code %d, for reason %q", configType, strconv.Itoa(code), output) 
}

// create the object if it doesn't exist yet
output, err = RunAdminCommandNoMultisite(objContext, false, args...)
if err != nil {
    return errorOrIsNotFound(err, "failed to create ceph %q %q, for reason %q", configType, configTypeArg, output))
}
logger.Debugf("created %q %q", configType, configTypeArg)

return nil
```


Finally, when we discover a bug, we should **always** create a test to make sure the error doesn't happen again. Please create a unit test for this function, and please add a test case for `createMultisite()` that ensures the behavior as well.